ieee802154/src/alif_mac154_parser.h

```diff
@@ -0,0 +1,241 @@
+#define MAC154_FRAME_VERSION_OFFSET 1
+#define MAC154_FRAME_VERSION_MASK   0x30
+#define MAC154_FRAME_VERSION_SHIFT  4
+
+/* Frame control field: Source addressing mode */
+#define MAC154_SOURCE_ADDRESSING_MODE_OFFSET 1
+#define MAC154_SOURCE_ADDRESSING_MODE_MASK   0xc0
+#define MAC154_SOURCE_ADDRESSING_MODE_SHIFT  6
+
+/* Secure control field */
+#define MAC154_SECURE_CONTROL_FIELD_SIZE                          1
+#define MAC154_SECURE_FRAME_COUNTER_SIZE                          4
+#define MAC154_SECURE_CONTROL_FIELD_FRAME_COUNTER_SUPPRESSION_BIT 0x20
+#define MAC154_SECURE_CONTROL_FIELD_SECURITY_LEVEL_MASK           0x07
+#define MAC154_SECURE_CONTROL_FIELD_KEY_IDENTIFIER_MODE_MASK      0x18
+#define MAC154_SECURE_CONTROL_FIELD_KEY_IDENTIFIER_MODE_SHIFT     3
+
+/* Get security enabled bit value from Frame control field.
+ */
+static inline bool alif_mac154_header_parser_security_enabled(const uint8_t *p_frame)
+{
+	return ((p_frame[MAC154_SECURITY_ENABLED_OFFSET] & MAC154_SECURITY_ENABLED_BIT) ==
+		MAC154_SECURITY_ENABLED_BIT);
+}
+
+/* Get Frame pending bit value from Frame control field.
+ */
+static inline bool alif_mac154_header_parser_frame_pending(const uint8_t *p_frame)
+{
+	return ((p_frame[MAC154_FRAME_PENDING_OFFSET] & MAC154_FRAME_PENDING_BIT) ==
+		MAC154_FRAME_PENDING_BIT);
+}
+
+/* Get acknowledge request bit value from Frame control field.
+ */
+static inline bool alif_mac154_header_parser_acknowledge_request(const uint8_t *p_frame)
+{
+	return ((p_frame[MAC154_ACKNOWLEDGE_REQUEST_OFFSET] & MAC154_ACKNOWLEDGE_REQUEST_BIT) ==
+		MAC154_ACKNOWLEDGE_REQUEST_BIT);
+}
+
+/* Get PANID compression bit value from Frame control field.
+ */
+static inline bool alif_mac154_header_parser_pan_id_compression(const uint8_t *p_frame)
+{
+	return ((p_frame[MAC154_PAN_ID_COMPRESSION_OFFSET] & MAC154_PAN_ID_COMPRESSION_BIT) ==
+		MAC154_PAN_ID_COMPRESSION_BIT);
+}
+
+/* Get Sequence number supression bit value from Frame control field.
+ */
+static inline bool alif_mac154_header_parser_seq_num_suppression(const uint8_t *p_frame)
+{
+	return ((p_frame[MAC154_SEQUENCE_NUMBER_SUPPRESSION_OFFSET] &
+		 MAC154_SEQUENCE_NUMBER_SUPPRESSION_BIT) == MAC154_SEQUENCE_NUMBER_SUPPRESSION_BIT);
+}
+
+/* Get information elements present bit value from Frame control field.
+ */
+static inline bool alif_mac154_header_parser_ie_present(const uint8_t *p_frame)
+{
+	return ((p_frame[MAC154_INFORMATION_ELEMENTS_PRESENT_OFFSET] &
+		 MAC154_INFORMATION_ELEMENTS_PRESENT_BIT) ==
+		MAC154_INFORMATION_ELEMENTS_PRESENT_BIT);
+}
+
+/* Get Frame type from Frame control field
+ *
+ * return values:
+ *      MAC154_FRAME_TYPE_BEACON         0x00
+ *      MAC154_FRAME_TYPE_DATA           0x01
+ *      MAC154_FRAME_TYPE_ACK            0x02
+ *      MAC154_FRAME_TYPE_COMMAND        0x03
+ *      MAC154_FRAME_TYPE_RESERVED       0x04
+ *      MAC154_FRAME_TYPE_MULTIPURPOSE   0x05
+ *      MAC154_FRAME_TYPE_FRAGMENT       0x06
+ *      MAC154_FRAME_TYPE_EXTENDED       0x07
```

Comment:
I wouldn't list the exact values and I would use doxygen formatting. But that's just my opinion.